tool/tsh/common/app_aws.go

```diff
@@ -168,36 +168,27 @@ func (a *awsApp) GetAppName() string {
 //
 // The first method is always preferred as the original hostname is preserved
 // through forward proxy.
-func (a *awsApp) StartLocalProxies() error {
-	// AWS endpoint URL mode
-	if a.cf.AWSEndpointURLMode {
-		if err := a.startLocalALPNProxy(a.cf.LocalProxyPort); err != nil {
-			return trace.Wrap(err)
-		}
+func (a *awsApp) StartLocalProxies(opts ...alpnproxy.LocalProxyConfigOpt) error {
+	ctx := context.TODO()
```

Comment:
I think context should be a parameter here (and in gcp/azure). Then callers can pass in the context from CLIConf which I think makes sense in most cases